skglm/utils.py

```diff
@@ -457,3 +457,114 @@ def extrapolate(self, w, Xw):
         C = inv_UTU_ones / np.sum(inv_UTU_ones)
         # floating point errors may cause w and Xw to disagree
         return self.arr_w_[:, 1:] @ C, self.arr_Xw_[:, 1:] @ C, True
+
+
+@njit
+def _prox_vec(w, z, penalty, lipschitz):
```

Comment:
rename lipschitz to step and use step=1/lipschitz. Otherwise this will become wrong if we decide to use linesearch at some point